### PR TITLE
The module search under add deadline deviations is case sensitive

### LINKED ISSUE #1280: The module search under Add Deadline Deviations is case sensitive
A minor usability issue:

When adding deadline deviations, I don’t suppose there is any reason for the Module selection to be case sensitive. 

In the pic below, I’m trying to locate a module named "Week 4" but "week" doesn’t find it.

image

### PR DESCRIPTION
**Pull request by ashi006**
Fixes #1280

# Description

**What?**

In add deadline deviations form, the module field search was case sensitive.

**Why?**

getSearchResults(() function in ajax_search_select.js was using the default :contains (which is case sensitive) selector to select the module value.

**How?**

Defined a custom selector "icontains" in ajax_search_results.js file to select elements that contain a specific text substring, case-insensitively and replaced :contains with :icontains in getSearchResults().

Fixes #1280 

# Testing

**Remember to add or update unit tests for new features and changes.**

* How to [test your changes in A-plus](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations)
* How to [test accessibility](https://wiki.aalto.fi/display/EDIT/How+to+check+the+accessibility+of+pull+requests)

**What type of test did you run?**

- [ ] Accessibility test using the [WAVE](https://wave.webaim.org/extension/) extension.
- [ ] Django unit tests.
- [ ] Selenium tests.
- [ ] Other test. *(Add a description below)*
- [x] Manual testing.

[ADD A DESCRIPTION ABOUT WHAT YOU TESTED MANUALLY]

**Did you test the changes in**

- [x] Chrome
- [ ] Firefox
- [ ] This pull request cannot be tested in the browser.

**Think of what is affected by these changes and could become broken**

# Translation

- [ ] Did you modify or add new strings in the user interface? ([Read about how to create translation](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations))

# Programming style

- [x] Did you follow our [style guides](https://apluslms.github.io/contribute/styleguides/)?
- [x] Did you use Python type hinting in all functions that you added or edited? ([type hints](https://docs.python.org/3/library/typing.html) for function parameters and return values)

# Have you updated the README or other relevant documentation?

- [ ] documents inside the doc directory.
- [ ] README.md.
- [ ] Aplus Manual.
- [ ] Other documentation (mention below which documentation).

# Is it Done?

- [ ] Reviewer has finished the code review
- [ ] After the review, the developer has made changes accordingly
- [ ] Customer/Teacher has accepted the implementation of the feature

*Clean up your git commit history before submitting the pull request!*

### CODE REVIEW

File: assets/js/ajax_search_select.js [hunk 1]
```diff
@@ -3,6 +3,10 @@ $(function() {
 
   $('.search-select').aplusSearchSelect();
   $('.search-select-ajax').aplusSearchSelectAjax();
+
+  jQuery.expr[':'].icontains = function(a, i, m) {
+    return jQuery(a).text().toUpperCase().indexOf(m[3].toUpperCase()) >= 0;
+  };
```
Comment (ihalaij1) [inline assets/js/ajax_search_select.js:9]: Is it possible to give more descriptive parameter names for `a`, `i`, and `m`?
Right now, I don't know what they mean just from looking at the code.

**Review by ihalaij1 — CHANGES_REQUESTED** [review]
Excellent!

See the single question that I had.

File: assets/js/ajax_search_select.js (hunk 1 above)
Comment (ashi006) [reply assets/js/ajax_search_select.js]: Yes, updated the code.

**Review by ihalaij1 — APPROVED** [review]

**Comment by ihalaij1** [comment]
Great, thank you!

**Pull request merged**